src/components/app/details/cdDetails/CDDetails.tsx

```diff
@@ -138,10 +138,12 @@ export default function CDDetails() {
             value: `${item.environmentId}`,
             label: item.environmentName,
             pipelineId: cdPipelinesMap.get(item.environmentId).id,
+            deploymentAppDeleteRequest: item.deploymentAppDeleteRequest,
         }
     })
 
-    if(result[0]['value'].result.length === 1 && !envId){
+    const isEnvDeleted = result[0]['value']?.result.find(_res => _res.deploymentAppDeleteRequest).deploymentAppDeleteRequest
+    if(result[0]['value'].result.length === 1 && !envId && !isEnvDeleted ){
```

Comment:
Does this require optional check if not then remove from 145 line as well or put it here

Reply:
Done